orc/src/main/java/org/apache/iceberg/orc/OrcMetrics.java

```diff
@@ -0,0 +1,162 @@
+import org.apache.iceberg.expressions.Literal;
+import org.apache.iceberg.hadoop.HadoopInputFile;
+import org.apache.iceberg.io.InputFile;
+import org.apache.iceberg.types.Conversions;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+import org.apache.orc.ColumnStatistics;
+import org.apache.orc.OrcFile;
+import org.apache.orc.OrcProto;
+import org.apache.orc.Reader;
+import org.apache.orc.TypeDescription;
+
+public class OrcMetrics {
+
+  private OrcMetrics() {}
+
+  public static Metrics fromInputFile(InputFile file) {
+    final Configuration config = (file instanceof HadoopInputFile)
+        ? ((HadoopInputFile)file).getConf()
+        : new Configuration();
+    return fromInputFile(file, config);
+  }
+
+  public static Metrics fromInputFile(InputFile file, Configuration config) {
+    try {
+      final Reader orcReader = OrcFile.createReader(new Path(file.location()),
+          OrcFile.readerOptions(config));
+      final TypeDescription orcSchema = orcReader.getSchema();
+      final Schema schema = TypeConversion.fromOrc(orcSchema);
+
+      ColumnStatistics[] colStats = orcReader.getStatistics();
+      List<OrcProto.ColumnStatistics> colStatsProto = orcReader.getOrcProtoFileStatistics();
+      Map<Integer, Long> columSizes = Maps.newHashMapWithExpectedSize(colStats.length);
+      Map<Integer, Long> valueCounts = Maps.newHashMapWithExpectedSize(colStats.length);
+      Map<Integer, Literal<?>> lowerBounds = Maps.newHashMap();
+      Map<Integer, Literal<?>> upperBounds = Maps.newHashMap();
+
+      for (int i = 0; i < colStats.length; i++) {
+        columSizes.put(i, colStats[i].getBytesOnDisk());
+        valueCounts.put(i, colStats[i].getNumberOfValues());
+
+        final OrcProto.ColumnStatistics protoStats = colStatsProto.get(i);
+        final Types.NestedField col = schema.findField(i);
+        if (col != null) {
+          Optional<Literal<?>> orcMin = fromOrcMin(col, protoStats);
+          if (orcMin.isPresent()) {
+            lowerBounds.put(i, orcMin.get());
+          }
+          Optional<Literal<?>> orcMax = fromOrcMax(col, protoStats);
+          if (orcMax.isPresent()) {
+            upperBounds.put(i, orcMax.get());
+          }
+        }
+      }
+
+      return new Metrics(orcReader.getNumberOfRows(),
+          columSizes,
+          valueCounts,
+          Maps.newHashMap(),
+          toBufferMap(schema, lowerBounds),
+          toBufferMap(schema, upperBounds));
+    } catch (IOException ioe) {
+      throw new RuntimeIOException(ioe, "Failed to read footer of file: %s", file);
+    }
+  }
+
+  private static Optional<Literal<?>> fromOrcMin(Types.NestedField column,
+                                                    OrcProto.ColumnStatistics columnStats) {
+    Literal<?> min = null;
+    if (columnStats.hasIntStatistics()) {
+      if (column.type().typeId() == Type.TypeID.INTEGER) {
+        min = Literal.of((int) columnStats.getIntStatistics().getMinimum());
```

Comment:
Why convert to bytebuffers in a separate function? It seems that if we do it here itself, we save traversing the columns again and possibly also simplify the code a little.
e.g instead of 
`Literal.of(columnStats.getDoubleStatistics().getMaximum());` we could do 
` Conversions.toByteBuffer(column.type(), columnStats.getDoubleStatistics().getMinimum());`

Reply:
Yeah, I think that should work. I'll try that. Thanks.